include/hmap/dynamic-hmap.hpp

```diff
@@ -173,11 +344,27 @@ class DynamicHMap {
 };
 
 template<typename V>
-detail::Key<V> dK(const std::string&k){
+detail::Key<V> dK(const std::string& k) {
 	return detail::Key<V>(k);
 }
 
+template<typename V>
+detail::Key<std::shared_ptr<V> > dSK(const std::string& k) {
+	return detail::Key<std::shared_ptr<V> >(k);
+}
+	
+template<typename V>
+detail::Key<std::unique_ptr<V> > dUK(const std::string& k) {
+	return detail::Key<std::unique_ptr<V> >(k);
+}
+	
 template<typename ...Vs>
 DynamicHMap make_dynamic_hmap(Vs&& ...vs) {
-	return DynamicHMap((std::initializer_list<std::pair<const detail::KeyBase, std::any>>){std::forward<Vs>(vs)...});
+	DynamicHMap hmap;
+	(static_cast<void>([&hmap](auto commaPair) {
+	                       hmap.map_.try_emplace(commaPair.first,
+	                                             std::move(commaPair.second));
+	                       }(vs)),
+	    ...);
```

Comment:
Is there any particular reason for this re-write? I usually assume that bulk construction will be faster than a sequence of inserts.

If we want to keep this form, it also probably makes sense to update the DynamcHMap class itself with try_emplace so other people can make use of it as well..